### ISSUE TITLE
AMP PyTorch 1.0.0 Support Question

### ISSUE THREAD

**Issue by kayuksel** [NVIDIA/apex#135]
Does **amp** support PyTorch **1.0.0**? Or shall I rather use the FP16_Optimizer? As far as my understanding, amp is a more advanced version of FP16_Optimizer which performs unsafe ops in FP32. The distinction and which one should be used is not super clear from the documentation but I guess that I will have to use FP16_Optimizer as amp doesn't seem to have full support for PyTorch 1.0.0 yet.

**Comment by kayuksel** [comment]
@mcarilli Hello, thank you very much for your response. I haven't tried amp with PyTorch 1.0 initially as the document stated 0.4.1 I have tried later but although it seems to be installed correctly (apex package is present when I do pip3 list or conda list), I am unable to import it (it says No module named amp). I have used FP16_Optimizer but that failed to reproduce the same accuracy. I have managed to run apex with PyTorch 0.4.1 and obtained memory reduction whilst the speed of the training does not seem to change a lot (only slightly). Thank you again for your time and consideration. Have a nice day. I will be looking forward for trying the unified API.

**Comment by mcarilli** [comment]
What's your import statement? Does `from apex import amp` not work?  Also, you should not import apex when you are in the top level apex repo directory, because then the pythonpath will initially attempt to import things directly from the files, rather than from the installed path.

You can also try running in a Docker container.  This will provide a controlled environment in which everything will *probably* work.  See https://github.com/NVIDIA/apex/issues/112#issuecomment-453307010.  For safety, choose a container labelled after the Cuda driver version that you have installed on bare metal.  For example, if your driver came as part of a cuda 10 install, use a cuda 10 container.

**Comment by kayuksel** [comment]
I guess it would work if I explicitly wrap local optimizers of hypertuning instances with a global amp handler.

**Comment by mcarilli** [comment]
The amp handler is always global, you should only ever need to call amp_handle = amp.init() once.

If you treat each optimizer as specified here https://github.com/NVIDIA/apex/tree/master/apex/amp#explicitly-wrapping-optimizers, dynamic loss scaling should be handled properly, independently, for each of them.
